framework/src/base/Assembly.C

```diff
@@ -2047,6 +2091,39 @@ Assembly::reinitLowerDElemRef(const Elem * elem,
   }
 }
 
+void
+Assembly::reinitLowerDElemDualRef(const Elem * elem,
+                                  const std::vector<Point> * const pts,
+                                  const std::vector<Real> * const weights)
+{
+  mooseAssert(pts->size(),
+              "Currently reinitialization of lower d elements is only supported with custom "
+              "quadrature points; there is no fall-back quadrature rule. Consequently make sure "
+              "you never try to use JxW coming from a fe_lower object unless you are also passing "
+              "a weights argument");
+
+  _current_lower_d_elem = elem;
+
+  unsigned int elem_dim = elem->dim();
+
+  for (const auto & it : _fe_lower[elem_dim])
+  {
+    FEBase * fe_lower = it.second;
+    FEType fe_type = it.first;
+    FEShapeData * fesd = _fe_shape_data_dual_lower[fe_type];
+
+    fe_lower->set_calculate_dual(true);
+    fe_lower->reinit(elem, pts, weights);
```

Comment:
@lindsayad here is where the dualLowerD elems are re-initialized with specific pts. `dual_coeff` is not computed in this case thus causes a seg fault...

Reply:
See my comment https://github.com/libMesh/libmesh/pull/2552#discussion_r423113037. Is it valid to calculate the dual coefficients with points that don't correspond to a quadrature rule for that element? Is the correct solution to call `fe_lower->reinit(elem)` first before calling `fe_lower->reinit(elem, pts, weights)`?

Reply:
The `reinitLowerDElemRef` and `reinitLowerDElemDualRef` are supposed to compute the quadrature rule from the customized integration points for the slave side. Apologies for the earlier confusion.. I also just realized that the `dual_coeff` should be computed using the same integration pts as the primal. So it is the "full" element for general cases, and becomes complicated for the slave elements when we are integrating using projected points  (`custom_xi1_pts`). And I guess the local matrices totally depend on the 'correctness' of the customized integration points in order to be non-singular..

Reply:
So what are you proposing? One can easily imagine (just thinking about the linear Lagrange case for simplicity) that you could have a mortar elem which occupies a very tiny fraction of the total slave element "volume", e.g. where the `custom_xi1_pts` are clustered on the farthest LHS or RHS of the slave element.

Reply:
My question to you is: is the dual method valid for any set of "integration" points, putting aside "practical" concerns like whether the biorthogonality condition yields a non-singular matrix?

Reply:
Theoretically it should be valid for the set of integration points that are the same of the primal element.

Reply:
Is "primal element" language that people use? What is a "primal element"?

Reply:
Good question. The "primal and dual" concept is something people often use. The "dual basis" is relative to the "primal basis". Here I think it is more appropriate to say the "element that uses the primal basis" and the "element that uses the dual basis"